03_Modules/Ocpi/src/apis/ChargingProfilesControllerApi.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  BaseOcpiHeaders,
+  setAuthHeader,
+  validateAndgetOcpiHeaders,
+} from './util';
+import { BaseAPI, HTTPHeaders } from './BaseApi';
+import { ChargingProfileResponse } from '../model/ChargingProfileResponse';
+import { OcpiResponse } from '@citrineos/base';
+import { VersionNumber } from '../../../../00_Base/src/interfaces/api/ocpi/model/VersionNumber';
+import { SetChargingProfile } from '../model/SetChargingProfile';
+
+export interface DeleteReceiverChargingProfileRequest extends BaseOcpiHeaders {
+  sessionId: string;
+  responseUrl: string;
+}
+
+export interface GetReceiverChargingProfileRequest extends BaseOcpiHeaders {
+  sessionId: string;
+  duration: number;
+  responseUrl: string;
+}
+
+export interface PutReceiverChargingProfileRequest extends BaseOcpiHeaders {
+  sessionId: string;
+  setChargingProfile: SetChargingProfile;
+}
+
+export class ChargingProfilesControllerApi extends BaseAPI {
+  async deleteReceiverChargingProfile(
+    requestParameters: DeleteReceiverChargingProfileRequest,
+    versionId: string = VersionNumber.TWO_DOT_TWO_DOT_ONE,
+  ): Promise<OcpiResponse<void>> {
+    BaseAPI.validateRequiredParam(
+      requestParameters,
+      'sessionId',
+      'responseUrl',
+    );
+
+    const queryParameters: any = {};
+
+    if (requestParameters.responseUrl != null) {
+      queryParameters['response_url'] = requestParameters.responseUrl;
+    }
+
+    const headerParameters: HTTPHeaders =
+      validateAndgetOcpiHeaders(requestParameters);
+
+    setAuthHeader(headerParameters);
+    return await this.request({
```

Comment:
created all Api classes to follow similar format to perform the `receiver` requests utilizing Base and helpers